includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -1038,6 +1116,66 @@ public function ajax_add_to_cart() {
 		wp_send_json( $data );
 	}
 
+	/**
+	 * Handles payment requests on the Pay for Order page.
+	 *
+	 * @throws Exception All exceptions are handled within the method.
+	 */
+	public function ajax_pay_for_order() {
+		check_ajax_referer( 'pay_for_order' );
+
+		if (
+			! isset( $_POST['payment_method'] ) || 'woocommerce_payments' !== $_POST['payment_method']
+			|| ! isset( $_POST['order'] ) || ! intval( $_POST['order'] )
+			|| ! isset( $_POST['wcpay-payment-method'] ) || empty( $_POST['wcpay-payment-method'] )
+		) {
+			// Incomplete request.
+			return;
+		}
+
+		try {
+			// Set up an environment, similar to core checkout.
+			wc_maybe_define_constant( 'WOOCOMMERCE_CHECKOUT', true );
+			wc_set_time_limit( 0 );
+
+			// Load the order.
+			$order_id = intval( $_POST['order'] );
+			$this->add_order_meta( $order_id );
+			$order = wc_get_order( $order_id );
+
+			if ( ! is_a( $order, WC_Order::class ) ) {
```

Comment:
Maybe we can move this check before `$this->add_order_meta( $order_id );` function to prevent strange behaviors in the codebase

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/4264/commits/54aadd28ed5b128424e317aa65a5a0e4c5169c9b